src/Streamly/Internal/FileSystem/Event/Linux.hs

```diff
@@ -0,0 +1,1058 @@
+
+-- | File/directory deleted from watched directory.
+--
+-- /Occurs only for an object inside the watched directory/
+--
+-- /Internal/
+--
+isDeleted :: Event -> Bool
+isDeleted = getFlag iN_DELETE
+
+-- XXX what if an object is moved in from outside or moved out of the monitored
+-- dir?
+--
+-- | Generated for the original path when an object is moved from under a
+-- monitored directory.
+--
+-- /Occurs only for an object inside the watched directory/
+--
+-- /Internal/
+--
+isMovedFrom :: Event -> Bool
+isMovedFrom = getFlag iN_MOVED_FROM
+
+-- | Generated for the new path when an object is moved under a monitored
+-- directory.
+--
+-- /Occurs only for an object inside the watched directory/
+--
+-- /Internal/
+--
+isMovedTo :: Event -> Bool
+isMovedTo = getFlag iN_MOVED_TO
+
+-- | Determine whether the event indicates modification of an object within the
+-- monitored path. This event is generated only for files and not directories.
+--
+-- /Occurs only for an object inside the watched directory/
+--
+-- /Internal/
+--
+isModified :: Event -> Bool
+isModified = getFlag iN_MODIFY
+
+-------------------------------------------------------------------------------
+-- Information about path type (applicable only when 'setFileEvents' is 'On')
+-------------------------------------------------------------------------------
+
+foreign import capi
+    "sys/inotify.h value IN_ISDIR" iN_ISDIR :: Word32
+
+-- | Determine whether the event is for a directory path.
+--
+-- /Internal/
+--
+isDir :: Event -> Bool
+isDir = getFlag iN_ISDIR
+
+-------------------------------------------------------------------------------
+-- Debugging
+-------------------------------------------------------------------------------
+
+-- | Convert an 'Event' record to a String representation.
+showEvent :: Event -> String
+showEvent ev@Event{..} =
+       "--------------------------"
+    ++ "\nWd = " ++ show eventWd
+    ++ "\nRoot = " ++ show (utf8ToString $ getRoot ev)
+    ++ "\nPath = " ++ show (utf8ToString $ getRelPath ev)
+    ++ "\nCookie = " ++ show (getCookie ev)
+    ++ "\nFlags " ++ show eventFlags
+
+    ++ showev isOverflow "Overflow"
+
+    ++ showev isRootUnwatched "RootUnwatched"
+    ++ showev isRootDeleted "RootDeleted"
+    ++ showev isRootMoved "RootMoved"
+    ++ showev isRootUnmounted "RootUnmounted"
+
+    ++ showev isMetadataChanged "MetadataChanged"
+
+    ++ showev isAccessed "Accessed"
+    ++ showev isOpened "Opened"
+    ++ showev isWriteClosed "WriteClosed"
+    ++ showev isNonWriteClosed "NonWriteClosed"
+
+    ++ showev isCreated "Created"
+    ++ showev isDeleted "Deleted"
+    ++ showev isModified "Modified"
+    ++ showev isMovedFrom "MovedFrom"
+    ++ showev isMovedTo "MovedTo"
+
+    ++ showev isDir "Dir"
+    ++ "\n"
+
+        where showev f str = if f ev then "\n" ++ str else ""
+#else
+#warning "Disabling module Streamly.Internal.FileSystem.Event.Linux. Does not support kernels older than 2.6.36."
```

Comment:
Do we get this warning when compiling on Darwin as well?

Reply:
The linux module is not compiled on Darwin, see the cabal file.